### PR TITLE
Test that IO.asyncPure terminates

### PR DESCRIPTION
**Pull request by artempyanykh**
I was on a 2 week old master and was bitten by https://github.com/scalaz/scalaz-zio/commit/8e55a4e46ec75eb933c51c2651226d1c6e1b7ac3

Took me awhile before figured it out. WDYT about adding a test on this?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 target
 .sbtopts
 project/.sbt
+.idea/
```
Comment (ktonga) [inline .gitignore]: Please remove these lines, it's developer's preference not project related

Comment (artempyanykh) [reply .gitignore]: @ktonga sure thing. Done!

**Comment by artempyanykh** [comment]
@ktonga ♻️

**Comment by ktonga** [comment]
Thank you @artempyanykh

**Pull request merged**